Assets/Scripts/GEGLevelController.cs

```diff
@@ -2,69 +2,132 @@
-    /// 
-    /// Q: Why is the dictBaseVals for each property not for each enemy type?
-    /// 
-    /// public Dictionary<string, float[]> UpdateProperties(int gameDiff, GEGPackedData packedData) { // new constructor
-    public Dictionary<string, float[]> UpdateProperties(int gameDiff, float[] typeDiff,
-        Dictionary<string, bool> dictProperty, Dictionary<string, int> dictBaseVals, Dictionary<string, bool> dictPropotion) {
+    public class GEGLevelController
+    {
+        List<Transform> enemySpawnPoints;
+        bool randomSpawn = true;
 
-        //float[] typeDiff = new float[packedData.enemyTypeData.Length];
-        //for (int i = 0; i < packedData.enemyTypeData.Length; ++i) {
-        //    // Construct what you need below
-        //}
+        public GEGLevelController()
+        {
+        }
 
-        Dictionary<string, float[]> propertyValues = new Dictionary<string, float[]>();
 
-        int numEnemyType = typeDiff.Length; // number of all enemy types
-        gameDiff /= 10; // convert difficulty level to percentage
+        //List<GEGTypeContainer> enemyTypeData;// = packedData.enemyTypeData;
+        /// <summary>
+        /// an example to run.
+        /// </summary>
+        /// <param name="difflevel">Difficulty level (from 0 to 10)</param>
+        /// <returns></returns>
+        public void RunExample(GEGPackedData data, int diffLevel)
+        {
+            List<int> enemys = enemyNumberGenerator(GEGPackedData.enemyTypeData, diffLevel);
+            enemypropertyGenerators(diffLevel, GEGPackedData.enemyTypeData);
+            List<List<int>> position = enemypositionsGenerator(enemys);
+            for (int i = 0; i < enemys.Count; i++)
+            {
+                for (int j = 0; j < enemys[i]; j++)
+                {
+                    Instantiate(GEGPackedData.enemyTypeData[i].prefab, enemySpawnPoints[position[i][j]].position);
+                }
 
-        // From endIndex to the end of array, the difficulty of those types should be 0 which means ignoring them
-        // Should use Mathf.Ceil to round up
-        int endIndex = (int)Mathf.Ceil(numEnemyType * gameDiff / 10); // Q: why not use gameDiffPercent?
-        for (int i = endIndex; i < numEnemyType; i++) {
-            typeDiff[i] = 0;
+            }
         }
 
-        // Loop properties that need to be modified
-        foreach (var item in dictProperty) {
-            if (item.Value == true) {
-                // The array containing values of the property for different types of enemies
-                float[] values = new float[numEnemyType];
+        /// <summary>
+        /// Generate enemy properties.
+        /// </summary>
+        /// <param name="difflevel">Difficulty level (from 0 to 10)</param>
+        /// <param name="PropertyList">Dictionary contains all attributes with enable or not</param>
+        /// <returns></returns>
+        void enemypropertyGenerators(int difflevel, List<GEGTypeContainer> enemyTypeData)
```

Comment:
- C# naming convention: Capitalize each word in function name => enemyPropertyGenerator